sympy/assumptions/assume.py

```diff
@@ -117,6 +119,20 @@ def binary_symbols(self):
                 return i.binary_symbols
         return set()
 
+class PolyadicAppliedPredicate(AppliedPredicate):
```

Comment:
I don't think we need a separate class. I would rather if AppliedPredicate just allowed multiple arguments.

Reply:
In fact, that's what it was in ce80e25. But it broke CNF-related logic which I couldn't fix - so I introduced this class to leave old parts alone. Can you help me fixing that?

Reply:
How does it break?